packages/ember-runtime/lib/system/core_object.js

```diff
@@ -393,6 +393,7 @@ CoreObject.PrototypeMixin = Mixin.create({
   _scheduledDestroy() {
     if (this.isDestroyed) { return; }
     destroy(this);
+    this.container = undefined;
```

Comment:
We should consider moving up the destroy queue in the run loop so that destroyed things don't remain reachable for long enough for this to matter.  I'm not sure why it is post render, this causes things to stay around when there is GC pressure.  I think we should move it up at least after the actions queue.